### ISSUE TITLE
index.d.ts type BoundAction: Expected 0 arguments, but got 1.

### ISSUE THREAD

**Issue by reallyely** [developit/unistore#129]
https://github.com/developit/unistore/blob/1df7cf60ac6fa1a70859d745fbaea7ea3f1b8d30/index.d.ts#L9

I'm looking into use cases for Unistore where I need to expose the store and bound actions for use in other frameworks.  I've been taking advantage of `store.action()` to get a handle on that bound action, but I get typing issues when that action takes parameters due to how `BoundAction` is defined. 

For example when I...
```ts
      // define the action as an unary function
      const actions = {
        toggleLoading: (_state: MyState, isLoading: boolean) => ({ isLoading })
      }
      // and then get the bound action from our previously created store
      const toggleLoading = store.action(actions.toggleLoading) 
      // I receive:  TS2554: Expected 0 arguments, but got 1.
```

I have temporarily sidestepped this issue by locally redefining `BoundAction` to be:
```ts
export type BoundAction = (...args: any) => void;
```

I'm a TypeScript newbie so I don't have good sensibilities when it comes to typing yet.  Is there a 'better' way to type this, or am I doing something wrong?

Thanks!

**Comment by developit** [comment]
@reallyely I've implemented your workaround in 6e26e4b5905b1757925cb587333867738178e3f7 for now.  I think there is probably a hacky way to get types to propagate here, but TypeScript doesn't have support for mixing Generics with variadic arguments, so it'd basically be listing out the possible argument types manually (gross).

**Comment by developit** [comment]
Released as [3.2.1](https://github.com/developit/unistore/releases/tag/3.2.1).
